extensions/arc/src/ui/tree/miaaTreeNode.ts

```diff
@@ -8,19 +8,19 @@ import * as vscode from 'vscode';
 import { ControllerModel } from '../../models/controllerModel';
 import { MiaaModel } from '../../models/miaaModel';
 import { MiaaDashboard } from '../dashboards/miaa/miaaDashboard';
-import { TreeNode } from './treeNode';
+import { ResourceTreeNode } from './resourceTreeNode';
 
 /**
  * The TreeNode for displaying a SQL Managed Instance on Azure Arc
  */
-export class MiaaTreeNode extends TreeNode {
+export class MiaaTreeNode extends ResourceTreeNode {
 
-	constructor(public model: MiaaModel, private _controllerModel: ControllerModel) {
-		super(model.info.name, vscode.TreeItemCollapsibleState.None, ResourceType.sqlManagedInstances);
+	constructor(model: MiaaModel, private _controllerModel: ControllerModel) {
+		super(model.info.name, vscode.TreeItemCollapsibleState.None, ResourceType.sqlManagedInstances, model);
 	}
 
 	public async openDashboard(): Promise<void> {
-		const miaaDashboard = new MiaaDashboard(this._controllerModel, this.model);
+		const miaaDashboard = new MiaaDashboard(this._controllerModel, this.model as MiaaModel);
```

Comment:
this.model is of type MiaaModel so why is the cast necessary? Perhaps add a comment to clarify